### DISCUSSION TITLE
Pull formulas

### DISCUSSION THREAD

**Discussion by jr212** [PHPOffice/PhpSpreadsheet#3847] [Ideas]
It will be nice if there is a feature that allows us to add the same formula into multiple cells at once of pull down/right/... like we can in Excel itself

Like "=A1" pull down 1 line becomes "=A2"
or 
$ws->setCellValue('A1:A2', '=B1'); so the formula in A2 = "=A2"

Jan

**Comment by oleibman** [comment]
Let me know if PR #3855 satisfies your requirement.

**Comment by jr212** [comment]
Please forgive me 3 times :)

1. Is this a solution already or just a request to do so? I don't know how to but it in working or do I have to compose it again
2. English is not my native language so not easy to understand
3. New to both composer and git

I'm happy that you will look at it and may have found a solution

Jan

**Reply by oleibman** [reply]
1. It is new code that I am considering adding to the project. It would be included with the next release or the one after that.
2. No problem. You've done well explaining what you want; I just want to be sure that I haven't overlooked something.
3. I can give you instructions on how to test the new code, but it may be too much for you if you are new to composer and git. Let me know if you want to try anyhow.

**Comment by jr212** [comment]
I will be happy to test. New to git/composer: yes

Almost 40 years of home programming (low level) and 20 in vba for the firm 😊

 

Van: oleibman ***@***.***> 
Verzonden: woensdag 3 januari 2024 16:12
Aan: PHPOffice/PhpSpreadsheet ***@***.***>
CC: jr212 ***@***.***>; Author ***@***.***>
Onderwerp: Re: [PHPOffice/PhpSpreadsheet] Pull formulas (Discussion #3847)

 

1.	It is new code that I am considering adding to the project. It would be included with the next release or the one after that.
2.	No problem. You've done well explaining what you want; I just want to be sure that I haven't overlooked something.
3.	I can give you instructions on how to test the new code, but it may be too much for you if you are new to composer and git. Let me know if you want to try anyhow.

**Comment by oleibman** [comment]
@jr212 You can get a version that includes this change via:
```
composer require phpoffice/phpspreadsheet:dev-master
```

**Reply by jr212** [reply]
tryed it :)
Recomposed without problems but how to use?

I get a error (Fatal error: Uncaught PhpOffice\PhpSpreadsheet\Exception: Cell coordinate string can not be a range of cells.) on code `$spreadsheet->setActiveSheetIndex(0)->setCellValue('F1:F5', 'Hello');`

What did I do?
remove the folder vendor completly
in commandscreen type the command you have givven
copy paste vendor to a test location
run file 01_Simple.php ==> OK
edited that file like I wrote above ==> NOK

**Comment by oleibman** [comment]
I could not think of a way to incorporate your idea into `setCellValue`. I added a new method `copyCells` instead, and a sample demonstrating how to use it. So your code should look something like:
```php
        $sheet->getCell('E3')->setValue('=A1&B1');
        $sheet->copyCells('E3', 'E3:F7');
```

**Comment by jr212** [comment]
Works fine :)
Just  an idee
    function multiCellsValue(string $toCells, string $Value, bool $copyStyle = true): void
    {
        $FirstCell = Coordinate::extractAllCellReferencesInRange($toCells)[0];
        $this->getCell($FirstCell)->setValue($value);
        copyCells(string $FirstCell, string $toCells, false);
    }

**Reply by andrz16-dev** [reply]
hi can you help me i cant using excel formula on my laravel project

**Comment by j212r** [comment]
Do you think it's a good idea to hijack someone else his topic?

create a new topic please.
You don't even provide info on whats the problem!
